hivemq-edge/src/main/java/com/hivemq/configuration/service/impl/ConfigurationServiceImpl.java

```diff
@@ -199,20 +199,20 @@ public void setConfigFileReaderWriter(final @NotNull ConfigFileReaderWriter conf
                     return method.invoke(instance, args);
                 } finally {
                     if(mutator){
-                        flush();
+                        flush(configFileReaderWriter);
                     }
                     presentLock.unlock();
```

Comment:
The flush-method has a try-block that doesn't catch.
So this might never be reached.